arrocke/gloss-translation
# feat: reset password

## What has changed
- Added a "Forgot Password?" button in the login page.
- Added two new pages:
  - `/forgot-password` - A page the user can email a link to themself which leads to `/reset-password`.
  - `/reset-password?token` - A page where the user submits their new password.
- Added two new api paths:
  - `POST /api/auth/forgot-password` - Creates a `ResetPasswordToken` from an email address passed in the body.
  - `POST /api/auth/reset-password` - Deletes the created `ResetPasswordToken`, updates the user's password, and logs the user in. 
- Created a schema in prisma for `ResetPasswordToken`




## Connected Issues

closes #373 


## QA Steps
1. Go to the Login page.
2. Press the "Forgot Password?" button. 
3. Enter "platform_admin@example.com" into the email. 
4. Press the "Reset Password" button. Confirm that a success message flashes.
5. If you have no email server set up, go to the api log and confirm that a reset password link has been sent. Go to that link.
6. Enter a new password and press "confirm" button.  Confirm that clicking the button logs you in. 
7. Log out. 
8. Confirm you can log back in with the new password.


## Post-Deployment
